Examples/MAX32670/Temp_Monitor/temp_monitor.c

```diff
@@ -0,0 +1,267 @@
+{
+    uint32_t temp_reading;
+    float temp_reading_float;
+
+    // Get current temperature
+    if (temp_sensor.read((void *)&temp_reading_float) != E_NO_ERROR) {
+        return;
+    }
+
+    // Record current temperature with timestamp
+    temp_reading = (uint32_t)temp_reading_float;
+    temp_readings[(num_readings % 4)] = (current_time << TIMESTAMP_OFFSET) | temp_reading;
+
+    // Increment readings counter (reset to 0 if flash page has been filled)
+    num_readings = (num_readings + 1) % (MXC_FLASH_PAGE_SIZE / sizeof(temp_readings[0]));
+
+    // Store temperatures to flash if 4 readings have been taken
+    // Note: we store them every 4 readings so we are able to do a 128-bit write
+    if (num_readings % 4 == 0) {
+        store_temp_readings();
+    }
+
+    // Check if temperature has exceeded a threshold
+    if (temp_reading > HI_TEMP_THRESHOLD) {
+        LED_Off(1);
+        temp_warning = true;
+        printf("Temperature exceeded upper threshold!\n");
+    } else if (temp_reading < LO_TEMP_THRESHOLD) {
+        LED_Off(1);
+        temp_warning = true;
+        printf("Temperature exceeded lower threshold!\n");
+    } else {
+        LED_Toggle(1); //Heartbeat
+        temp_warning = false;
+    }
+
+    return;
+}
+
+/***** RTC Functions *****/
+static int start_rtc(void)
+{
+    int err;
+
+    // Initialize RTC
+    if ((err = MXC_RTC_Init(0, 0)) != E_NO_ERROR) {
+        return err;
+    }
+
+    // Setup TOD alarm
+    MXC_RTC_DisableInt(MXC_RTC_INT_EN_LONG); //TOD alarm interrupt most be disabled to set period
+    if ((err = MXC_RTC_SetTimeofdayAlarm(TEMP_CHECK_PERIOD)) != E_NO_ERROR) {
+        return err;
+    }
+    MXC_RTC_EnableInt(MXC_RTC_INT_EN_LONG);
+
+    // Enable RTC Wakeup
+    MXC_LP_EnableRTCAlarmWakeup();
+    NVIC_EnableIRQ(RTC_IRQn);
+
+    // Start RTC
+    return MXC_RTC_Start();
+}
+
+static void enable_warning_light(void)
+{
+    // Set subsecond alarm period and enable interrupt
+    MXC_RTC_DisableInt(MXC_RTC_INT_EN_SHORT); //SSEC alarm must be disabled to set period
+    MXC_RTC_SetSubsecondAlarm(WARN_LIGHT_PERIOD);
+    MXC_RTC_EnableInt(MXC_RTC_INT_EN_SHORT);
+    return;
+}
+
+static void disable_warning_light(void)
+{
+    // Disable SSEC alarm
+    MXC_RTC_DisableInt(MXC_RTC_INT_EN_SHORT);
+
+    LED_Off(0); //Make sure warning light is off
+    return;
+}
+
+/******************************************************************/
+/************************ Public Functions ************************/
+/******************************************************************/
+
+/***** Initialization Function *****/
+int temp_monitor_init(void)
+{
+    int err;
+
+    if ((err = init_flash()) != E_NO_ERROR) { // Initialize flash space used to store temp readings
+        return err;
+    } else if ((err = init_temp_sensor()) !=
+               E_NO_ERROR) { // Initialize temperature sensor (and I2C)
+        return err;
+    } else if ((err = start_rtc()) !=
```

Comment:
Weird clang-format. Put comment inside of `if else` body so the line doesn't get wrapped.